### PR TITLE
Implement get_value_type

### LINKED ISSUE #11: Implement get_value_type
This for each should take a variable number of arguments and apply a type trait to all of them. I don't know the interface for this.

### PR DESCRIPTION
**Pull request by Zanthelo**
Closes #11

### CODE REVIEW

**Comment by Highlife1911** [comment]
Add a static assert in Parse<first> in a new function that gets selected when there is no such id that triggers a static assert that says that the ID is not found.

**Comment by Highlife1911** [comment]
Can you change this to a trait that returns the index of the type and not the type? This is more helpful because we are using tuples to store the Connectors and with the index we can get the type and the value.

File: source/yats/include/yats/Util.h [hunk 1]
```diff
@@ -69,4 +69,61 @@ struct TaskHelper
...
+	{
+		return outer_parse_index<Index + 1>();
+	}
+
+	using type = decltype(decltype(Help(std::declval<T>()))::parse_type());
+	static constexpr uint64_t index = outer_parse_index();
+};
+
+template<uint64_t Id, typename T>
+using get_element_by_id_t = typename get_element_by_id<Id, T>::type;
```
Comment (Highlife1911) [inline source/yats/include/yats/Util.h]: v

**Comment by lawben** [comment]
To be honest, I don't understand all of this. IF @Highlife1911 approves, so do I :)

File: source/yats/include/yats/Util.h (hunk 1 above)
Comment (Zanthelo) [reply source/yats/include/yats/Util.h]: If we follow the style of the standard library it should be _t since it is a type.

Comment (Highlife1911) [reply source/yats/include/yats/Util.h]: but std::tuple_element_v is also a type and uses v

**Pull request merged**